PHP/PHP Source.sublime-syntax

```diff
@@ -811,748 +795,839 @@ contexts:
-    - match: '(\$+){{identifier}}'
-      scope: variable.parameter.php
+      pop: 1
+    - match: (\$){{identifier}}
+      scope: variable.other.php
       captures:
         1: punctuation.definition.variable.php
-    - include: punctuation-separators
 
-  function-return-type:
-    - meta_scope: meta.function.return-type.php
-    - include: expression-end-pop
-    - match: \{
-      scope: meta.block.php punctuation.section.block.begin.php
-      set: function-body
-    - match: ':'
-      scope: punctuation.separator.colon.php
-    - include: type-hints
-    # Exit on unexpected content
-    - match: (?=\S)
+###[ FOR STATEMENTS ]#########################################################
+
+  for-statements:
+    - match: (?i:for)\b
+      scope: keyword.control.loop.for.php
+      push: for-statement-group
+
+  for-statement-group:
+    - match: \(
+      scope: punctuation.section.group.begin.php
+      set: for-statement-group-body
+    - include: comments
+    - include: else-pop
+
+  for-statement-group-body:
+    - meta_scope: meta.group.php
+    - match: \)
+      scope: punctuation.section.group.end.php
+      pop: 1
+    - match: (?=}|\s*\?>|{{keywords}})
       pop: 1
+    - include: expressions
+    - include: punctuation-terminators
 
-  function-body:
-    - meta_scope: meta.function.php meta.block.php
+###[ USE STATEMENTS ]#########################################################
+
+  use-statements:
+    - match: (?i:use)\b
+      scope: keyword.declaration.use.php
+      push: use-statement-body
+
+  use-statement-body:
+    - meta_scope: meta.use.php
+    - include: comments
+    - match: (?i:const)\b
+      scope: keyword.declaration.constant.php
+      set: use-statement-const
+    - match: (?i:function)\b
+      scope: keyword.declaration.function.php
+      set: use-statement-function
+    - match: (?=\S)
+      set: use-statement-ns-class
+
+
+  use-statement-const:
+    - meta_content_scope: meta.use.php
+    - match: \{
+      scope: punctuation.section.sequence.begin.php
+      push: use-statement-const-body
+    - include: use-statement-const-common
+
+  use-statement-const-body:
+    - meta_scope: meta.sequence.list.php
     - match: \}
-      scope: punctuation.section.block.end.php
+      scope: punctuation.section.sequence.end.php
       pop: 1
-    - include: embedded-html
+    - include: use-statement-const-common
+
+  use-statement-const-common:
+    - match: (?i:as)\b
+      scope: keyword.other.as.php
+      push: use-statement-const-alias
+    - include: use-statement-common
+    - include: punctuation-separators
+    # Don't mark this as entity.name.constant since the name is pulled verbatim
+    # and the indexer should find the name in the original source file.
+    - include: constants
+
+  use-statement-const-alias:
+    - match: '{{guarded_identifier}}'
+      scope: entity.name.constant.php
+      pop: 1
+    - include: attributes
+    - include: comments
+    - include: else-pop
+
```

Comment:
```suggestion
```

Reply:
Those duplicated empty lines are used to group the different types of contexts. They exist by intent

Reply:
My bad 👍🏻